### ISSUE TITLE
Implement a language server for Fusion

### ISSUE THREAD

**Issue by iahung2** [fusionlanguage/fut#126]
### Discussed in https://github.com/fusionlanguage/fut/discussions/124

There is no existing issue for this, so I created a new one to track the progress.

**Comment by ariankordi** [comment]
+1 on this, having to use VS Code for the plugin as a non-user is inconvenient.

I should also ask how multi-file projects are going to work with this. As of now there's no "import" system with Fusion, but the VS Code extension does seem to read multiple files anyway?: https://github.com/fusionlanguage/fut/blob/09b63732e5fa19a23baabc42a899b01b891750ad/editors/vscode/src/extension.ts#L56-L67
In my experience I've seen this work sometimes if I'm opening one folder with only .fu files, but for a repo with multiple levels and many .fu files, then it doesn't pick up on any other files.

**Comment by pfusik** [comment]
Indeed, there's no `import` statement. You need to pass all the files on the fut command line. The VS Code extension assumes the .fu source files are contained in the very same folder (no subfolders).
